src/account_manager.rs

```diff
@@ -495,19 +500,26 @@ async fn sync_accounts<'a>(
         synced_accounts.push(synced_account);
     }
 
-    let discovered_accounts = match last_account {
+    let discovered_accounts_res = match last_account {
         Some(account) => {
             if account.messages().is_empty()
                 || account.addresses().iter().all(|addr| *addr.balance() == 0)
             {
-                discover_accounts(&storage_path, account.client_options()).await?
+                discover_accounts(
+                    &storage_path,
+                    account.client_options(),
+                    Some(account.signer_type().clone()),
+                )
+                .await
             } else {
-                vec![]
+                Ok(vec![])
```

Comment:
I'ld use `Vec::new()` instead of `vec![]`. Let's say later on we decide to use a `HashMap` here, or use some other construction method of the `Vec` like `with_capacity`, then there's a bit less to change :smile: `Vec::new` is also easier to read IMHO.

Reply:
usually I go with vec![] but you're right, i'll update all references on a subsequent PR